lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -374,9 +374,25 @@ def full_replacement_map
                          @define_variables)
     end
 
+    def read_dynamic_device(device)
+      begin
+        require Pathname.new(DEVICES_JSON_DIR).join("#{device}.rb").realpath.to_s
```

Comment:
We shouldn't rescue `require`; `require` may raise `Errno::ENOENT` because of the code written in the required file.